sql/parser/eval.go

```diff
@@ -795,7 +795,8 @@ func evalFuncExpr(expr *FuncExpr) (Datum, error) {
 		types = append(types, reflect.TypeOf(arg))
 	}
 
-	if expr.fn.fn == nil {
+	// Aggregate functions get looked up fresh every time.
```

Comment:
Yep. Basically, `COUNT(*)` will get looked up initially as `COUNT(tuple)` but then executed as `COUNT(int)`. Maybe there is a better way to handle this while still keeping the argument type checking.